src/atc/configurator/configurator.py

```diff
@@ -39,57 +52,103 @@ def __init__(self, resource_path: Union[str, ModuleType] = None):
+        # tc._get_item("MyLiteral") -> "MyValue"
+        # tc._get_item("MyKey") -> {'name':..., 'path':...}
+        # tc._get_item("MyAlias") -> same as tc._get_item("MyKey")
+        # tc._get_item("MyForked")
+        #   if not self._is_debug -> same as tc._get_item("MyKey")
+        #   if self._is_debug -> {'name':'anotherName', 'path':'another/Path'}
+        #
+        # resolving forks and aliases is the task of this method.
+
+        # this stack allows us to detect alias loops
         stack = {table_id}
 
-        value: TcDetails = self._raw_resource_details[table_id]
-
-        while set(value.keys()) in [{"release", "debug"}, {"alias"}]:
-            if set(value.keys()) == {"release", "debug"}:
-                # Handle the case of differentiated release and debug tables
-                if self._is_debug:
-                    value = value["debug"]
+        value: TcValue = self._raw_resource_details[table_id]
+
+        while True:
+            if not isinstance(value, dict):
+                # situation like MyLiteral
+                return value
+            else:
+                # value is a dict
+                if set(value.keys()) == {"release", "debug"}:
+                    # Situation like MyForked
+                    if self._is_debug:
+                        value = value["debug"]
+                    else:
+                        value = value["release"]
+                    continue
+                elif set(value.keys()) == {"alias"}:  # allow alias of alias
+                    # Situation like MyAlias
+                    new_id = value["alias"]
+                    if new_id in stack:
+                        raise ValueError(f"Alias loop at key {new_id}")
+                    stack.add(new_id)
+                    value = self._raw_resource_details[new_id]
+                    continue
                 else:
-                    value = value["release"]
-
-            if set(value.keys()) == {"alias"}:  # allow alias of alias
-                new_id = value["alias"]
-                if new_id in stack:
-                    raise ValueError(f"Alias loop at key {new_id}")
-                stack.add(new_id)
-                value = self._raw_resource_details[new_id]
-
-        # we are finished resolving a possible route of aliases
-        return value
+                    # value is a dict,
+                    # but it is neither a forking (release,debug)
+                    # nor is it an alias
+                    # hence we have arrived at the final value set of this item
+                    # This is how both MyForked and MyAlias can arrive at MyKey
+                    return value
 
     def _get_unsubstituted_item_property(self, table_id: str, property: str) -> str:
         """item property where release-debug and alias loops are resolved"""
         value = self._get_item(table_id)
-
+        if not property:
```

Comment:
there will always be a property since is is mandatory input to the method.

Reply:
"if not property" will always be false

Reply:
Not true, you can actively supply an empty string which will be false. And this is exactly what I do.